stdio/sprintf.c

```diff
@@ -18,28 +18,38 @@
 #include "format.h"
 
 
+typedef struct _vsprintf_ctx_t {
+	char *buff;
+	size_t n;
+} vsprintf_ctx_t;
+
+
 typedef struct _vsnprintf_ctx_t {
 	char *buff;
 	size_t n;
 	size_t max_len;
 } vsnprintf_ctx_t;
 
 
-static void vsnprintf_feed(void *context, char c)
+static void vsprintf_feed(void *context, char c)
 {
-	vsnprintf_ctx_t* ctx = (vsnprintf_ctx_t *)context;
+	vsprintf_ctx_t* ctx = (vsprintf_ctx_t *)context;
 
-	if (ctx->max_len == 0 || ctx->n + 1 < ctx->max_len) {
-		ctx->buff[ctx->n++] = c;
+	ctx->buff[ctx->n++] = c;
+}
 
-		if (ctx->max_len && ctx->n + 1 == ctx->max_len)
-			ctx->buff[ctx->n] = '\0';
 
-	} else {
-		/* Count anyway to return the number of characters that would have been
-		 * written if buffer had been sufficiently large */
-		ctx->n++;
+static void vsnprintf_feed(void *context, char c)
+{
+	vsnprintf_ctx_t* ctx = (vsnprintf_ctx_t *)context;
+
+	if (ctx->n < ctx->max_len) {
+		ctx->buff[ctx->n] = c;
 	}
+
+	/* Count anyway to return the number of characters that would have been
+	 * written if buffer had been sufficiently large */
+	ctx->n++;
```

Comment:
according to C99 the `snprintf` output should always be null-terminated if `max_len > 0` (https://en.cppreference.com/w/c/io/fprintf).

I think You removed this functionality in case the result string overflows the buffer.